lib/rules/no-shared-component-data.js

```diff
@@ -21,10 +27,33 @@ function create (context) {
         p.value.type !== 'ArrowFunctionExpression' &&
         p.value.type !== 'Identifier'
       )
-      .forEach(cp => {
+      .forEach(p => {
         context.report({
-          node: cp.value,
-          message: '`data` property in component must be a function'
+          node: p,
+          message: '`data` property in component must be a function',
+          fix (fixer) {
+            let first = sourceCode.getFirstToken(p.value)
+            let last = sourceCode.getLastToken(p.value)
+
+            // If the value enclosed by parentheses, update the 'first' and 'last' by the parentheses.
+            let t = null
```

Comment:
I think we should use more descriptive variables, `t` and `u` doesn't mean much.